### PR TITLE
Feature branch sync up with main branch

### PR DESCRIPTION
**Pull request by RyanL1997**
### Description
Feature branch sync up with main

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: gradle/formatting.gradle [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            }
+        }
+        format 'misc', {
+            target '*.md', '*.gradle', '**/*.json', '**/*.yaml', '**/*.yml', '**/*.svg'
+
+            trimTrailingWhitespace()
+            endWithNewline()
+        }
+    }
+}
```
Comment (scrawfor99) [inline gradle/formatting.gradle:36]:
```suggestion
}

```

File: formatter/formatterConfig.xml [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+        <setting id="org.eclipse.jdt.core.formatter.tabulation.char" value="space"/>
+        <setting id="org.eclipse.jdt.core.formatter.insert_space_after_relational_operator" value="insert"/>
+        <setting id="org.eclipse.jdt.core.formatter.insert_space_before_comma_in_multiple_field_declarations" value="do not insert"/>
+        <setting id="org.eclipse.jdt.core.formatter.wrap_before_string_concatenation" value="true"/>
+        <setting id="org.eclipse.jdt.core.formatter.blank_lines_between_import_groups" value="1"/>
+        <setting id="org.eclipse.jdt.core.formatter.lineSplit" value="140"/>
+        <setting id="org.eclipse.jdt.core.formatter.insert_space_after_opening_paren_in_annotation" value="do not insert"/>
+        <setting id="org.eclipse.jdt.core.formatter.insert_space_before_opening_paren_in_switch" value="insert"/>
+    </profile>
+</profiles>
```
Comment (scrawfor99) [inline formatter/formatterConfig.xml:362]:
```suggestion
</profiles>

```

**Comment by DarshitChanpura** [comment]
Are we still pushing changes to this branch?

**Comment by cwperks** [comment]
@RyanL1997 I am seeing some differences after I did a merge from main -> feature/extensions and resolved conflicts. The file count on this https://github.com/opensearch-project/security/compare/feature/extensions...cwperks:format-feature-extensions-apply?expand=1 is different from this PR.

I checked out your branch and did a merge of main -> this branch and had to resolve the same conflicts from merging main into the branch in the link above.

**Comment by RyanL1997** [comment]
> Are we still pushing changes to this branch?

Hi @DarshitChanpura. Yes this [PR](https://github.com/opensearch-project/security/pull/2833) needs the formatting changes + library fix to run the test I think

**Review by peternied — CHANGES_REQUESTED** [review]
Lets please hold off on these changes , I've still got outstanding commits for this branch that I'd like to land before we pull in all the style changes.

**Review by peternied — COMMENTED** [review]
I'm no longer impacted by this changes getting merged.  Please use `rebase and merge` instead of `squash and merge` via the GitHub UX so it makes it easier to manage the changes.

**Comment by RyanL1997** [comment]
Closing this one due to the merge of: https://github.com/opensearch-project/security/pull/2854

**Pull request closed without merging**